engines/macventure/macventure.cpp

```diff
@@ -0,0 +1,1184 @@
+						_paused = true;
+					} else {
+						_paused = false;
+						updateState(true);
+						updateControls();
+						updateExits();
+					}
+				}
+
+				if (_gameState == kGameStateWinnig || _gameState == kGameStateLosing) {
+					endGame();
+				}
+			}
+		}
+		refreshScreen();
+	}
+
+	return Common::kNoError;
+}
+
+void MacVentureEngine::refreshScreen() {
+	_gui->draw();
+	g_system->updateScreen();
+	g_system->delayMillis(50);
+}
+
+void MacVentureEngine::newGame() {
+	_world->startNewGame();
+	reset();
+	setInitialFlags();
+	setNewGameState();
+}
+
+void MacVentureEngine::setInitialFlags() {
+	_paused = false;
+	_halted = false;
+	_cmdReady = false;
+	_haltedAtEnd = false;
+	_haltedInSelection = false;
+	_clickToContinue = true;
+	_gameState = kGameStateInit;
+	_destObject = 0;
+	_prepared = true;
+}
+
+void MacVentureEngine::setNewGameState() {
+	_cmdReady = true;
+	ObjID playerParent = _world->getObjAttr(1, kAttrParentObject);
+	_currentSelection.push_back(playerParent);// Push the parent of the player
+	_world->setObjAttr(playerParent, kAttrContainerOpen, 1);
+}
+
+void MacVentureEngine::reset() {
+	resetInternals();
+	resetGui();
+}
+
+void MacVentureEngine::resetInternals() {
+	_scriptEngine->reset();
+	_currentSelection.clear();
+	_objQueue.clear();
+	_textQueue.clear();
+}
+
+void MacVentureEngine::resetGui() {
+	_gui->reloadInternals();
+	_gui->updateWindowInfo(kMainGameWindow, getParent(1), _world->getChildren(getParent(1), true));
+	// HACK! should update all inventories
+	_gui->ensureInventoryOpen(kInventoryStart, 1);
+	_gui->updateWindowInfo(kInventoryStart, 1, _world->getChildren(1, true));
+	updateControls();
+	updateExits();
+	refreshScreen();
+}
+
+void MacVentureEngine::requestQuit() {
+	// TODO: Display save game dialog and such
+	_gameState = kGameStateQuitting;
+}
+
+void MacVentureEngine::requestUnpause() {
+	_paused = false;
+	_gameState = kGameStatePlaying;
+}
+
+void MacVentureEngine::selectControl(ControlAction id) {
+	debugC(2, kMVDebugMain, "Select control %x", id);
+	if (id == kClickToContinue) {
+		_clickToContinue = false;
+		_paused = true;
+		return;
+	}
+
+	_selectedControl = id;
+	refreshReady();
+}
+
+void MacVentureEngine::refreshReady() {
+	switch (getInvolvedObjects()) {
+	case 0: // No selected object
+		_cmdReady = true;
+		break;
+	case 1: // We have some selected object
+		_cmdReady = _currentSelection.size() != 0;
+		break;
+	case 2:
+		if (_destObject > 0) // We have a destination seleted
+			_cmdReady = true;
+		break;
+	}
+}
+
+void MacVentureEngine::preparedToRun() {
+	_prepared = true;
+}
+
+void MacVentureEngine::gameChanged() {
+	_gameChanged = true;
+}
+
+void MacVentureEngine::winGame() {
+	_gui->showPrebuiltDialog(kWinGameDialog);
+	_gameState = kGameStateWinnig;
+}
+
+void MacVentureEngine::loseGame() {
+	_gui->showPrebuiltDialog(kLoseGameDialog);
+	_paused = true;
+	//_gameState = kGameStateLosing;
+}
+
+void MacVentureEngine::clickToContinue() {
+	_clickToContinue = true;
```

Comment:
I can't see it in the flow of the program, but in any case this function can and should be used to initialize it, since the engine starts requesting a click to continue.